core-bundle/src/Resources/config/listener.yml

```diff
@@ -329,6 +329,12 @@ services:
             - { name: kernel.event_listener, priority: -256 }
 
     contao.listener.tl_content_undo_description:
-        class: Contao\CoreBundle\EventListener\Undo\ContentUndoDescriptionListener
+        class: Contao\CoreBundle\EventListener\Undo\SingleTableInheritanceUndoDescriptionListener
+        # TODO: Complete field map
+        # TODO: Also make field map extendible
+        arguments:
+            - 'tl_content'
+            - 'type'
+            - { headline: 'headline', hyperlink: 'url', image: 'singleSRC', gallery: 'multiSRC' }
```

Comment:
This listener handles Single Table Inheritance data containers like `tl_content`: It takes the table name (`tl_content`), the discriminator column (`type`) and a field map as arguments. The field map maps the object name (`hyperlink` for the hyperlink element to the field, that describes this object best (The `url` field for the  `hyperlink` element).

I can provide a complete list for the CoreBundle and hard-code it here. But 1) is this the right place or it should be in another config file? And 2) How could this be extendible? I don't want developers to register this service a second time with their own field map, just because they added an content element.

Reply:
Another approach could be to mark a field as description column in the DCA field definition, e.g.
```
'url' => [
// ...
'descriptionFor' => 'hyperlink' // Headline field is the description field for the row if the discriminiator column equals 'headline'
]
```

This way we could gather the information from the already merged DCA and inject it to the listener. This way we do not add another way of defining something related to the DCA and it would automatically be extendeble as well.